### ISSUE TITLE
[Feature Request] Support simple native variable declarations/assignments

### ISSUE THREAD

**Issue by beccasaurus** [googleapis/sample-tester#53]
```yaml
setup:
  variable: { name: test_bucket, value: horse-sunset-4789 }

  variable:
    name: video_path
    value: gs://the-bucket/path/to/a/reused/value

  variable: { name: string_variable, value: "1234" }
  variable: { name: integer_variable, value: 1234 }
  variable: { name: float_variable, value: 12.34 }
  variable: { name: boolean_variable, value: true }
  variable: { name: list_variable: value: [1,2,3,4] }
  variable: { name: dictionary_variable: value: { 1: 2, 3: 4 } }

  # tl;dr simply rely on YAML deserialization / variable inflation, no special handling
```

I keep finding myself wanting to re-use variables.

Or simply add some variables to setup to things like long GCS paths, so I can use them from test cases.

Using `code: x=y` is not an option, I'm trying to not use `code:` unless absolutely necessary :)

**Comment by beccasaurus** [comment]
Note: variables cannot reference other variables, other local variables in the scope, or invoke any methods. Just natives as deserialized by YAML.
